cfme/tests/containers/test_alerts.py

```diff
@@ -1,20 +1,23 @@
 import pytest
 
 from cfme.containers.provider import ContainersProvider
+from cfme.markers.env_markers.provider import providers
 from cfme.utils.version import current_version
+from cfme.utils.providers import ProviderFilter
 
 
 pytestmark = [
     pytest.mark.uncollectif(lambda: current_version() < "5.9"),
-    pytest.mark.provider([ContainersProvider], scope='module')
+    pytest.mark.provider(gen_func=providers,
+                         filters=[ProviderFilter(classes=[ContainersProvider],
+                                                 required_flags=['prometheus_alerts'])],
+                         scope='function')
 ]
 
+# TODO There needs to be more to this test
 
-@pytest.fixture
-def ensure_alerts(provider):
-    if 'alerts' not in provider.endpoints:
-        pytest.skip('No alerts endpoint found for this provider')
 
-
-def test_add_alerts_provider(ensure_alerts, provider):
+@pytest.mark.uncollectif(lambda provider: 'alerts' not in provider.endpoints,
```

Comment:
DataProvider PR will be merged today. Once it is merged, provider fixture will hold DataProvider instead of BaseProvider.  If you really need to check whether provider supports alerts, you need to add provider marker with required_fields set.